andela/codepirates-ah-backend
# #167313396 Build out Admin REST endpoints for resources(CRUD)

#### What does this PR do?
- Have super admin created automatically
- Have admin endpoints built out 
- Have the admin's endpoints tested
#### Description of Task to be completed?
- Implemented TDD practice
- Protected Admin Routes
- Implemented role based Authorizations
- CRUD operations on admin 
 Have The following endpoint working and tested
*   POST/localhost/api/v1/users/login
*   POST/localhost/api/v1/users/signup
*   GET/localhost/api/v1/users/allusers
*   GET/localhost /api/v1/users/1
*   DELETE/localhost/api/v1/users/5
*   POST/localhost/api/v1/users/signup/admin
* PUT/ocalhost/api/v1/users/update/xxx@gmail.com
#### How should this be manually tested?
* Download or clone the tutorial project code from [this repo](https://github.com/andela/codepirates-ah-backend.git),Install all required NPM  packages by running NPM install from the command line in the project root folder (where the package.json is located).
* Start the API by running NPM  start from the command line in the project root folder, you should see the message Server listening on port 8000. You can test the API directly using an application such as Postman or you can test it with one of the single pages in the browser.
* to test authentication and authorization, add this to the header 
Key: :x-access-token,value:JWT TOKEN
#### Any background context you want to provide?

#### What are the relevant pivotal tracker stories?
[#167313421](https://www.pivotaltracker.com/n/projects/2372121/stories/167313421)

#### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/16412410/62596366-48805380-b8e2-11e9-9963-9eef660e91ea.png)

#### Questions:
